### ISSUE TITLE
Example is out of date and confusing

### ISSUE THREAD

**Issue by tksuoran** [MicrosoftDocs/cpp-docs#3773]
Example on this page is out of date and confusing. While the example may have been valid in previous versions of Visual Studio, now that the default fp contract mode was changed, the comment in the example does not really apply.

---
#### Document Details

⚠ *Do not edit this section. It is required for docs.microsoft.com ➟ GitHub issue linking.*

* ID: 2d02db96-cc68-3aa1-9eef-068f72115f2c
* Version Independent ID: f9b0f598-d877-26c6-75aa-52d016c150a0
* Content: [fp_contract pragma](https://docs.microsoft.com/en-us/cpp/preprocessor/fp-contract?view=msvc-170)
* Content Source: [docs/preprocessor/fp-contract.md](https://github.com/Microsoft/cpp-docs/blob/main/docs/preprocessor/fp-contract.md)
* Product: **visual-cpp**
* Technology: **cpp-language**
* GitHub Login: @corob-msft
* Microsoft Alias: **corob**

**Comment by corob-msft** [comment]
@tksuoran 

There are subtleties here. The example tells you to compile under `/fp:fast`, which implicitly enables contractions. The `#pragma fp_contract(off)` line disables contractions unconditionally. If you comment it out as suggested, then the line in the description is still true in any version of Visual Studio: "the generated code may use a fused-multiply-add instruction if it is available." This matches what I observed when I tried the example on godbolt.org using the `x64 msvc v19.latest` compiler with options `/W4 /std:c++17 /O2 /fp:fast /arch:AVX2`. The output used `vmulsd` followed by `vaddsd` with the pragma, and `vfmadd231sd` without the pragma.

So, I don't quite agree with your assessment on being out of date, but I could be convinced about the confusing part. We don't clarify the behavior of `/fp:fast` here, and it has a material effect on the example.
